### ISSUE TITLE
Issue in connection containing '#'

### ISSUE THREAD

**Issue by laedit** [r3c/creep#2]
I am trying to use creep with an ftp connection, but I have the following error:

``` bash
[1;37mINFO: Deploying to location "default"...[0m
Traceback (most recent call last):
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\man-p\AppData\Local\Programs\Python\Python35\Scripts\creep.exe\__main__.py", line 9, in <module>
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\site-packages\creep\creep.py", line 41, in main
    args.yes):
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\site-packages\creep\src\deploy.py", line 60, in execute
    if not sync (logger, base_path, definition, location, name, append_files, remove_files, rev_from, rev_to, yes):
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\site-packages\creep\src\deploy.py", line 109, in sync
    data = target.read (logger, location.state)
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\site-packages\creep\src\targets\ftp.py", line 50, in read
    ftp = self.connect (logger)
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\site-packages\creep\src\targets\ftp.py", line 22, in connect
    ftp.connect (self.host, self.port)
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\ftplib.py", line 153, in connect
    source_address=self.source_address)
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\socket.py", line 693, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "c:\users\man-p\appdata\local\programs\python\python35\lib\socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 11001] getaddrinfo failed
````

It appear that the [connection regex](https://github.com/r3c/creep/blob/ad7199dfe666e7b988c58e9cc51be929ef3af37d/creep/src/factory.py#L27) does not allow the `#` char, but my password contains one.

Is it possible to remove this restriction?

**Comment by r3c** [comment]
Hello, a plain # at this position is incorrect indeed, you should encode it as "%23" to get it accepted.

**Comment by laedit** [comment]
Thanks, I haven't realized that it must be url encoded.
But I ran into another issue:
``` bash
AttributeError: module 'urllib' has no attribute 'unquote_plus'
```

It seems that in Python 3 it is [`urllib.parse.unquote_plus`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote_plus).

**Comment by r3c** [comment]
I believe this issue is now fixed thanks to your suggestions from #3 ; feel free to reopen if you notice anything wrong here.
